Makefile

```diff
@@ -190,7 +194,14 @@ BINDIR = $(BUILD_PATH)/bin
 # ###################################################################
 
 L3_SRC      := $(L3_SRCDIR)/l3.c
-L3_ASSEMBLY := l3.S
+
+# Currently, fast-logging assembly is only supported on x86 64-bit Linux.
+ifeq ($(UNAME_S),Linux)
+    ifeq ($(UNAME_P), x86_64)
+        L3_ASSEMBLY := l3.S
```

Comment:
@gregthelaw - How hard is it to support assembly for MacOSX? 

I heard there are two diff chips Mac runs on: Intel and something else. Are you comfortable doing this work?

If so, I'd like to log an issue requesting this support, so it can be scheduled sometime in your busy schedule.

Reply:
Newer Macs have ARM CPUs. It's not fundamentally difficult to support, but I'm not sure there's a huge benefit to doing so. The fast logging was just to get a sense of what a hand-crafted assembly version would perform like. It's not that much faster than the C version.

More troublingly, I've realised there is a problem with the assembly code we have -- if linked against programs that are also using TLS, it will probably conflict. I need to figure out how to craft the assembly in the right way for TLS accesses. This probably isn't too hard, but does need doing.

Reply:
Meanwhile, I suggest we live without the fast version on architectures other than x86-64.